src/main/java/net/dries007/tfc/common/blocks/crop/CropBlock.java

```diff
@@ -158,7 +160,10 @@ public void tick(BlockState state, ServerLevel level, BlockPos pos, Random rand)
         {
             if (canSurvive(state, level, pos))
             {
-                level.getBlockEntity(pos, TFCBlockEntities.CROP.get()).ifPresent(crop -> growthTick(level, pos, state, crop));
+                if (level.getBlockEntity(pos) instanceof CropBlockEntity crop)
+                {
+                    growthTick(level, pos, state, crop);
+                }
```

Comment:
As a thought - instead of switching all these optional usages to `instanceof` - to allow addons with their own block entity types, could we prefer instead just replacing the explicit block entity type with `getExtendedProperties().blockEntity()` ? It's a bit neater, and it preserves the intended use (restricting based on the block entity type)

I think that still solves the issue at hand and doesn't require refactoring other usages of `getBlockEntity` as much.

Reply:
Ship may have sailed on this for some cases, but sure

Reply:
Not saying we need to refactor every usage we currently have, for sure. But for cases like these, and for new code - we can still use the block entity type that already exists on the block.